enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+Scylla Operator is configured to create Headless Services.
+
+![Private-VPC+Public-SNI-Proxy](private-vpc-public-sni-proxy.svg "Private VPC + Public SNI Proxy")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: PodIP
+```
+
+---
+
+#### Multicloud
+
+In multicloud type we rely on any solution providing inter-VPC connectivity. Two clouds need to have distinct VPC
+subnets with extra routes to point traffic to the second cloud over VPN. Scylla's nodes talk over VPN with each other,
+where clients talk through SNI proxy either for traffic originating from the same VPC, Private Link connection or public
+Internet.
+Scylla Operator is configured to create Headless Services.
+
+![Multicloud](multicloud.svg "Multicloud")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: PodIP
+```
+
+---
+
+## API changes
+
+Existing ExposeOptions - `scyllacluster.spec.exposeOptions` - will be extended with new fields:
+
+```go
+
+type PodIPSourceType string
+
+const (
+    // StatusPodIPSource specifies that the PodIP is taken from Pod.Status.PodIP
+    StatusPodIPSource PodIPSourceType = "Status"
+    
+    // InterfacePodIPSource specifies the PodIP is taken from the interface.
+    InterfacePodIPSource PodIPSourceType = "Interface"
+)
+
+type PodIPInterfaceOptions struct {
+    // interfaceName specifies interface name within a Pod from which address is taken from.
+    InterfaceName string `json:"interfaceName,omitempty"`
+}
+
+// PodIPAddressOptions hold options related to Pod IP address.
+type PodIPAddressOptions struct {
+    // source specifies source of the Pod IP.
+    // +kubebuilder:default:="Status"
+    Source PodIPSourceType `json:"source"`
+    
+    // interfaceOptions hold options for Interface SourceType.
+    // +optional
+    InterfaceOptions *PodIPInterfaceOptions `json:"interfaceOptions,omitempty"`
+}
+
+type BroadcastAddressType string
+
+const (
+    // PodIPBroadcastAddressType selects the IP address from Pod.
+    PodIPBroadcastAddressType BroadcastAddressType = "PodIP"
+    
+    // ServiceClusterIPBroadcastAddressType selects the IP address from Service.spec.ClusterIP
+    ServiceClusterIPBroadcastAddressType BroadcastAddressType = "ServiceClusterIP"
+    
+    // ServiceLoadBalancerIngressIPBroadcastAddressType selects the IP address from Service.status.ingress[0].ip
+    ServiceLoadBalancerIngressIPBroadcastAddressType BroadcastAddressType = "ServiceLoadBalancerIngressIP"
+)
+
+// BroadcastOptions hold options related to address broadcasted by ScyllaDB node.
+type BroadcastOptions struct {
+    // type of broadcasted address.
+    Type BroadcastAddressType `json:"type"`
+    
+    // podIP hold options related to Pod IP address.
+    // +optional
+    PodIP *PodIPAddressOptions `json:"podIP"`
+}
+
+// NodeBroadcastOptions hold options related to addresses broadcasted by ScyllaDB node.
```

Comment:
BroadcastOptions and NodeBroadcastOptions have the exact same comment

Reply:
not exactly the same, but i improved it.